six/modules/c++/cphd/include/cphd/SupportBlock.h

```diff
@@ -41,70 +41,112 @@
 //  any actual SupportBlock data
 class SupportBlock
 {
 public:
-    static const std::string ALL;
     /*
-     * \param pathname Input CPHD pathname
-     * \param data Data section from CPHD
-     * \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
-     * \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
+     *  \func SupportBlock
+     *
+     *  \brief Constructor initializes book keeping information
+     *
+     *  \param pathname Input CPHD pathname to initialize a file input stream
+     *  \param data Data section from CPHD
+     *  \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
+     *  \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
      */
     SupportBlock(const std::string& pathname,
                  const cphd::Data& data,
                  sys::Off_T startSupport,
                  sys::Off_T sizeSupport);
 
     /*
-     * \param inStream Input stream to an already opened CPHD file
-     * \param data Data section from CPHD
-     * \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
-     * \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
+     *  \func SupportBlock
+     *
+     *  \brief Constructor initializes book keeping information
+     *
+     *  \param inStream Input stream to an already opened CPHD file
+     *  \param data Data section from CPHD
+     *  \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
+     *  \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
      */
     SupportBlock(mem::SharedPtr<io::SeekableInputStream> inStream,
                  const cphd::Data& data,
                  sys::Off_T startSupport,
                  sys::Off_T sizeSupport);
 
     /*
-     * Return offset from start of CPHD file for a support array with
-     * specified id
-     * \param id unique id of support array
+     *  \func getFileOffset
+     *
+     *  \brief Get the byte offset of a specific support array in the CPHD file
+     *
+     *  \param id Unique id of support array
+     *
+     *  \return Returns offset from start of CPHD file
      */
     sys::Off_T getFileOffset(const std::string& id) const;
 
     /*
-     * Read the specified support array
-     * Performs endian swapping if necessary
-     *
-     * \param id unique identifier of support array
-     * \param numThreads Number of threads to use for endian swapping if
-     * necessary
-     * \param data Will contain the read in data.  Throws if buffer has not
-     * been allocated to a sufficient size
-     * (numRows * numCols * bytesPerElement)
+     *  \func read
+     *
+     *  \brief Read the specified support array.
+     *
+     *   Performs endian swapping if necessary
+     *
+     *  \param id unique identifier of support array
+     *  \param numThreads Number of threads to use for endian swapping if
+     *   necessary
+     *  \param data A pre allocated mem::BufferView that will hold the data
```

Comment:
`\param[in,out]` since it's also an output parameter

Reply:
Do I need to do this for all parameters or just the input/output ambiguous ones?

Reply:
Just for input/output and output parameters